app/src/main/java/com/example/task/adapters/PendingAdapter.kt

```diff
@@ -0,0 +1,54 @@
+package com.example.task.adapters
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.PopupMenu
+import android.widget.TextView
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.recyclerview.widget.RecyclerView
+import com.example.task.MainActivity
+import com.example.task.R
+import com.example.task.database.Task
+import com.example.task.models.TaskViewModel
+
+class PendingAdapter(private val taskList: List<Task>,private val lambdaMarkCompleted: ()->TaskViewModel) : RecyclerView.Adapter<PendingAdapter.PendingViewHolder>(){
+    class PendingViewHolder(itemView: View): RecyclerView.ViewHolder(itemView){
+        val textView: TextView = itemView.findViewById(R.id.listItemTextField)
+        val listItem: ConstraintLayout = itemView.findViewById(R.id.list_item)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PendingViewHolder {
+        val view = LayoutInflater.from(parent.context).inflate(R.layout.task_list,parent,false)
+        return PendingViewHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: PendingViewHolder, position: Int) {
+        holder.textView.text = taskList[position].task
+        holder.listItem.setOnClickListener {
+            //lambda implementation should be there i.e sharedViewModel should not be used here
+            lambdaMarkCompleted.invoke().markCompleted(taskList[position])
```

Comment:
This doesn't make much sense. Ideally you should pass a simple onClick lambda that should be called whenever the item is clicked, and provide the lambda's body in the fragment.

Example:

```kotlin

// Fragment code
binding.recyclerView.adapter(taskList) { 
    viewModel.markComplete(task)
}

// Adapter code
class Adapter(..., private val onClick: (Task) -> Unit) : ... {
...
taskItem.setOnClickListener {
    onClick(taskList[position])
}

Reply:
Notice that we're passing the task object as a parameter to the lambda function.

Reply:
I am not using the onClick lambda because i need the flexibility to use the delete and edit feature too